nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/service/ColdStorageServiceImpl.java

```diff
@@ -54,84 +75,121 @@
 
-    private static final Logger log = LogManager.getLogger(ColdStorageHelper.class);
+    private static final Logger log = LogManager.getLogger(ColdStorageServiceImpl.class);
 
-    public static final String COLD_STORAGE_FACET_NAME = "ColdStorage";
-
-    public static final String FILE_CONTENT_PROPERTY = "file:content";
-
-    public static final String COLD_STORAGE_CONTENT_PROPERTY = "coldstorage:coldContent";
-
-    public static final String COLD_STORAGE_BEING_RETRIEVED_PROPERTY = "coldstorage:beingRetrieved";
+    protected static final List<String> COLD_STORAGE_DISABLED_RECOMPUTATION_LISTENERS = Arrays.asList(
+            UpdateThumbnailListener.THUMBNAIL_UPDATED, ThumbnailConstants.DISABLE_THUMBNAIL_COMPUTATION,
+            VideoChangedListener.DISABLE_VIDEO_CONVERSIONS_GENERATION_LISTENER,
+            PictureViewsGenerationListener.DISABLE_PICTURE_VIEWS_GENERATION_LISTENER);
 
-    /** @since 10.10 **/
-    public static final String COLD_STORAGE_TO_BE_RESTORED_PROPERTY = "coldstorage:toBeRestored";
+    public static final String COLDSTORAGE_RENDITION_EP = "coldStorageRendition";
 
-    public static final String GET_DOCUMENTS_TO_CHECK_QUERY = String.format(
-            "SELECT * FROM Document, Relation WHERE %s = 1", COLD_STORAGE_BEING_RETRIEVED_PROPERTY);
+    protected static String defaultRendition;
 
-    /** @since 10.10 **/
-    public static final String COLD_STORAGE_CONTENT_RESTORED_EVENT_NAME = "coldStorageContentRestored";
+    protected static Map<String, String> renditionByDocType = new HashMap<>();
 
-    /** @since 10.10 **/
-    public static final String COLD_STORAGE_CONTENT_TO_RESTORE_EVENT_NAME = "coldStorageContentToRestore";
+    protected static Map<String, String> renditionByFacets = new HashMap<>();
 
-    public static final String COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME = "coldStorageContentAvailable";
+    public ColdStorageServiceImpl() {
+        // no instance allowed
+    }
 
-    /** @since 10.10 **/
-    public static final String COLD_STORAGE_CONTENT_MOVED_EVENT_NAME = "coldStorageContentMoved";
+    @Override
+    public void start(ComponentContext context) {
+        List< ColdStorageRenditionDescriptor> descriptors = getDescriptors(COLDSTORAGE_RENDITION_EP);
+        descriptors.forEach(descriptor -> {
+            String docType = descriptor.getDocType();
+            if (docType != null) {
+                renditionByDocType.put(docType, descriptor.getRenditionName());
+            }
+            String facet = descriptor.getFacet();
+            if (facet != null) {
+                renditionByFacets.put(facet, descriptor.getRenditionName());
+            }
+            if (docType == null && facet == null) {
+                defaultRendition = descriptor.getRenditionName();
+                if (defaultRendition == null) {
+                    throw new NuxeoException("Please contribute a default rendition name");
```

Comment:
You could add the contribution name: `String.format("Please contribute a default rendition name: %s", descriptor.getName())`

Reply:
Thanks  a lot,
I will also extract `descriptor.getRenditionName() ` as a variable.
(already the case for 11.x and 2021.x)